grammar.ebnf

```diff
@@ -36,11 +42,37 @@ Statement = "{", {Statement}, "}"
 */
 LValue = Identifier, [ ".", Identifier ];
 
-FuncCallArgList = [ Expression, {",", Expression} ];
+Expression = ExpressionTerm 
+           | "(", Expression, ")"
+           | FunctionCall
+           | "!", Expression
+           | ExpressionTerm, "+", ExpressionTerm
+           | ExpressionTerm, "-", ExpressionTerm 
+           | ExpressionTerm, "*", ExpressionTerm
+           | ExpressionTerm, "/", ExpressionTerm
+           | ExpressionTerm, "%", ExpressionTerm
+           | ExpressionTerm, "[", ExpressionTerm, "]"
+           | ExpressionTerm, "&&", ExpressionTerm
+           | ExpressionTerm, "||", ExpressionTerm
+           | ExpressionTerm, "<", ExpressionTerm
+           | ExpressionTerm, ">", ExpressionTerm
+           | ExpressionTerm, "=<", ExpressionTerm
+           | ExpressionTerm, ">=", ExpressionTerm
+           | ExpressionTerm, "==", ExpressionTerm
+           | ExpressionTerm, "!=", ExpressionTerm;
```

Comment:
You can compress all of those with something like:
```
BinOp = "+" | "-" | ...
BinaryExpression = ExpressionTerm, BinOp, ExpressionTerm
```
Then add the `BinaryExpression` as an Expression`

Reply:
Fair enough.